### PR TITLE
gh-187: Support environment variables PYTHON_GIL / PYTHON_CPU_COUNT

### PR DESCRIPTION
**Pull request by corona10**
* https://docs.python.org/3.13/using/cmdline.html#envvar-PYTHON_CPU_COUNT
* https://docs.python.org/3.13/using/cmdline.html#envvar-PYTHON_GIL

### CODE REVIEW

**Comment by corona10** [comment]
cc @tonybaloney

**Comment by corona10** [comment]
@hugovk  PTAL for this PR too :)

**Review by hugovk — APPROVED** [review]

File: pyperf/_utils.py [hunk 1]
```diff
@@ -266,8 +266,10 @@ def create_environ(inherit_environ, locale, copy_all):
     if copy_all:
         return os.environ
     env = {}
-
-    copy_env = ["PATH", "PYTHONPATH", "HOME", "TEMP", "COMSPEC", "SystemRoot", "SystemDrive"]
+    copy_env = ["PATH", "HOME", "TEMP", "COMSPEC", "SystemRoot", "SystemDrive"]
+    # TODO: In the future, maybe we should manage CPython environment variables
+    # depending on the Python version.
+    copy_env.extend(["PYTHONPATH", "PYTHON_CPU_COUNT", "PYTHON_GIL"])
```
Comment (vstinner) [inline pyperf/_utils.py:272]: Just put it directly in the list, no?

File: pyperf/_utils.py [hunk 2]
```diff
@@ -266,8 +266,10 @@ def create_environ(inherit_environ, locale, copy_all):
     if copy_all:
         return os.environ
     env = {}
-
-    copy_env = ["PATH", "PYTHONPATH", "HOME", "TEMP", "COMSPEC", "SystemRoot", "SystemDrive"]
+    copy_env = ["PATH", "HOME", "TEMP", "COMSPEC", "SystemRoot", "SystemDrive"]
+    # TODO: In the future, maybe we should manage CPython environment variables
```
Comment (vstinner) [inline pyperf/_utils.py:270]: There is no need to check the Python version. You can remove the TODO.

File: pyperf/_utils.py (hunk 1 above)
Comment (corona10) [reply pyperf/_utils.py:272]: Well let's separate CPython specific variables :)

Comment (vstinner) [reply pyperf/_utils.py:271]: You can separate them with a comment and/or a newline.

Comment (corona10) [reply pyperf/_utils.py:271]: Well, I am not sure why you want to maintain the single list declaration even if this is not a performance-sensitive code

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -22,32 +22,54 @@ jobs:
       - run: tox -e ${{ matrix.tox-job }}
   build:
     runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }} - ${{ matrix.python }} ${{ matrix.build }}
```
Comment (vstinner) [inline .github/workflows/build.yml:25]: The PR title is only about "env var", but these changes look unrelated to env vars. You might merge them in a separated PR.

**Comment by vstinner** [comment]
Remark: If clearing env vars is a surprising/bad default behavior for end users, we can also just remove it? I did it to make benchmarks easier to reproduce.

**Comment by corona10** [comment]
>Remark: If clearing env vars is a surprising/bad default behavior for end users, we can also just remove it? I did it to make benchmarks easier to reproduce.

I think that the propagation is the more precise way. I prefer to support it as same as currently.

File: .github/workflows/build.yml (hunk 3 above)
Comment (corona10) [reply .github/workflows/build.yml]: Done

**Review by vstinner — APPROVED** [review]
LGTM
